src/memray/reporters/assets/temporal_flamegraph.js

```diff
@@ -51,6 +50,88 @@ function packedDataToTree(packedData, rangeStart, rangeEnd) {
     node["children"] = node["children"].map((idx) => node_objects[idx]);
   }
 
+  return node_objects;
+}
+
+function initTrees(packedData) {
+  const {
+    strings,
+    nodes,
+    inverted_no_imports_nodes,
+    unique_threads,
+    intervals,
+    no_imports_interval_list,
+  } = packedData;
+  const flamegraphNodeObjects = generateNodeObjects({
+    strings: strings,
+    nodes: nodes,
+    unique_threads: unique_threads,
+  });
+
+  const invertedNoImportsNodeObjects = inverted
+    ? generateNodeObjects({
+        strings: strings,
+        nodes: inverted_no_imports_nodes,
+        unique_threads: unique_threads,
+      })
+    : null;
+
+  flamegraphIntervals = intervals;
+  invertedNoImportsIntervals = no_imports_interval_list;
+
+  return {
+    flamegraphNodeObjects: flamegraphNodeObjects,
+    invertedNoImportsNodeObjects: invertedNoImportsNodeObjects,
+  };
+}
+
+function findHWMAllocations(intervals, node_objects, hwmSnapshot) {
+  if (!node_objects) {
+    return;
+  }
+
+  intervals.forEach((interval) => {
+    let [allocBefore, deallocBefore, nodeIndex, count, bytes] = interval;
+
+    if (
+      allocBefore <= hwmSnapshot &&
+      (deallocBefore === null || deallocBefore > hwmSnapshot)
+    ) {
+      while (nodeIndex !== undefined) {
+        node_objects[nodeIndex].n_allocations += count;
+        node_objects[nodeIndex].value += bytes;
+        nodeIndex = parent_index_by_child_index[nodeIndex];
+      }
+    }
+  });
+}
+
+function findLeakedAllocations(intervals, node_objects, rangeStart, rangeEnd) {
+  if (!node_objects) {
+    return;
+  }
+
+  intervals.forEach((interval) => {
+    let [allocBefore, deallocBefore, nodeIndex, count, bytes] = interval;
+
+    if (
+      allocBefore >= rangeStart &&
+      allocBefore <= rangeEnd &&
+      (deallocBefore === null || deallocBefore > rangeEnd)
+    ) {
+      while (nodeIndex !== undefined) {
+        node_objects[nodeIndex].n_allocations += count;
+        node_objects[nodeIndex].value += bytes;
+        nodeIndex = parent_index_by_child_index[nodeIndex];
```

Comment:
Ah, this is the bug you hit. You're starting with an index into `inverted_no_imports_nodes` and looking it up in `parent_index_by_child_index`, which is a mapping from index in `nodes` to index in `nodes`.

`parent_index_by_child_index` only knows about the main tree, not the inverted-no-import-system-nodes tree. We need a separate mapping table that maps child node index to parent node index for the new tree.